docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+######################
+User Interface Design
+######################
+
+This document outlines the design principles and guidelines for the user
+interface (UI) of NWChemEx. Our main goal is to provide a user-friendly and
+intuitive interface that enables users to carry out a variety of quantum
+chemistry calculations seamlessly. 
+
+
+Design Principles
+=================
+
+Performant
+-----------
+
+Performance is critical for NWChemEx with kernels supporting optimized MPI
+parallelization and GPU offloading. The UI is designed to uphold this
+performance standard and to provide the user with critical control elements to
+achieve high performance. This requires keeping the overhead of UI as minimal as
+possible by avoiding unnecessary copies or data movements and enabling user to
+access and modify the MPI communicator and GPU offloading as needed.
```

Comment:
I am thinking of having simple boolean control switches in `options` like `enable_gpu_integrals`, `enable_gpu_ccsd(t)` etc. Top layer functions in UI will load the right submodules based on these switches. I don't think this would require any change in ParallelZone.